src/jackmd/scorefactory/ScoreFactory.php

```diff
@@ -10,48 +10,33 @@
 use pocketmine\Player;
 use BadFunctionCallException;
 use OutOfBoundsException;
-use function mb_strtolower;
+use function array_map;
+use function array_values;
 
 class ScoreFactory{
 
-	/** @var string */
 	private const OBJECTIVE_NAME = "objective";
-	/** @var string */
 	private const CRITERIA_NAME = "dummy";
 
-	/** @var int */
 	private const MIN_LINES = 1;
-	/** @var int */
 	private const MAX_LINES = 15;
 
-	/** @var int */
 	public const SORT_ASCENDING = 0;
-	/** @var int */
 	public const SORT_DESCENDING = 1;
 
-	/** @var string */
 	public const SLOT_LIST = "list";
-	/** @var string */
 	public const SLOT_SIDEBAR = "sidebar";
-	/** @var string */
 	public const SLOT_BELOW_NAME = "belowname";
 
-	/** @var string[] */
-	private static array $scoreboards = [];
+	/** @var ScoreCache[] */
+	private static array $cache = [];
 
 	/**
 	 * Adds a Scoreboard to the player if he doesn't have one.
 	 * Can also be used to update a scoreboard.
-	 *
-	 * @param Player $player
-	 * @param string $displayName
-	 * @param int    $slotOrder
-	 * @param string $displaySlot
-	 * @param string $objectiveName
-	 * @param string $criteriaName
 	 */
 	public static function setScore(Player $player, string $displayName, int $slotOrder = self::SORT_ASCENDING, string $displaySlot = self::SLOT_SIDEBAR, string $objectiveName = self::OBJECTIVE_NAME, string $criteriaName = self::CRITERIA_NAME): void{
-		if(isset(self::$scoreboards[mb_strtolower($player->getName())])){
+		if(isset(self::$cache[$player->getRawUniqueId()])){
 			self::removeScore($player);
```

Comment:
Maybe we should stop resetting the scoreboard by default? I would think that setScore should only be called to initialize the scoreboard and then maybe an updateScore to actually update the scoreboard? It's not a big deal, but should help reduce the packet spam

Reply:
You are right.. but then again, lines/entries on the scoreboard won't update unless it's removed and added again and thus this check. I tried but was unable to get the scoreboard to update without removing it first. 
However, if you can manage to get it to update without removing it first then this could be removed, other than that, I think that it should stay.